numpyro/util.py

```diff
@@ -392,6 +392,11 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
     ys = lax.map(fn, xs) if num_chunks > 1 else fn(xs)
     map_ndims = int(num_chunks > 1) + int(chunk_size > 1)
     ys = tree_map(
-        lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
+        lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size]
+        if y.size
+        else y,
+        ys,
+    )
+    return tree_map(
+        lambda y: jnp.reshape(y, batch_shape + jnp.shape(y)[1:]) if y.size else y, ys
```

Comment:
Assuming that batch_shape=(2, 3, 4) and ys.shape = (24, 0), then we need to return `(2, 3, 4, 0)` instead of `(24, 0)`. Could you address this issue and help me add a regression test similar to [this one](https://github.com/pyro-ppl/numpyro/blob/master/test/test_util.py#L93) with
```
def f(x):
    return jnp.zeros((0,))
```